presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1143,6 +1120,88 @@ protected Scope visitTable(Table table, Optional<Scope> scope)
+             *  The optimizer should convert this query to
+             *  select a_c, b_c from mv_table where c=1 UNION ALL select a*c, b*c from base_table where c=1;
+             */
+
+            //select a*c as a_c, b*c as b_c, c from base_table
+            String mvCreateSql = view.getOriginalSql();
+            Statement mvCreateSqlStatement = sqlParser.createStatement(mvCreateSql, createParsingOptions(session, warningCollector));
+            QuerySpecification mvCreateSqlSpecification = (QuerySpecification) ((Query) mvCreateSqlStatement).getQueryBody();
+            Query mvCreateSqlQuery = new Query(Optional.empty(), mvCreateSqlSpecification, Optional.empty(), Optional.empty());
+
+            QuerySpecification userQuerySpecification = (QuerySpecification) ((Query) statement).getQueryBody(); //select a_c,b_c from mv_table where c=1
+
+            //select * from mv_table
+            QuerySpecification newQueryMVPart = new QuerySpecification(
+                    selectList(new AllColumns()),
+                    userQuerySpecification.getFrom(),
+                    Optional.empty(),
+                    Optional.empty(),
+                    Optional.empty(),
+                    Optional.empty(),
+                    Optional.empty()
+            );
+            Query newQueryMV = new Query(Optional.empty(), newQueryMVPart, Optional.empty(), Optional.empty());
+
+            //select * from mv_table UNION ALL select a*c as a_c, b*c as b_c, c from base_table
+            Union union = new Union(ImmutableList.of(mvCreateSqlQuery.getQueryBody(), newQueryMV.getQueryBody()), Optional.of(Boolean.FALSE));
+            Query unionQuery = new Query(Optional.empty(), union, Optional.empty(), Optional.empty());
+
+            //select a_c,b_c from (select * from mv_table UNION ALL select a*c as a_c, b*c as b_c, c from base_table) where c=1;
+            Query q = simpleQuery(
+                    userQuerySpecification.getSelect(),
+                    subquery(unionQuery),
+                    Optional.empty(),
+                    Optional.empty(),
+                    Optional.empty(),
+                    Optional.empty(),
+                    Optional.empty()
+            );
+
+            String newSql = SqlFormatterUtil.getFormattedSql(q, sqlParser, Optional.empty());
+            System.out.println("New generated SQL is: \n" + newSql);
+
+            Query query = parseView(newSql, name, table);
+
+            analysis.registerNamedQuery(table, query);
+
+            analysis.registerTableForMV(table);
+            RelationType descriptor = analyzeView(query, name, view.getCatalog(), Optional.of(view.getSchema()), view.getOwner(), table);
+            analysis.unregisterTableForMV(table);
+
+            RelationType outputDescriptor = analysis.getOutputDescriptor(query);
+            Collection<Field> visibleFields = outputDescriptor.getVisibleFields();
+
+            List<ViewDefinition.ViewColumn> columns = visibleFields.stream()
```

Comment:
Given query `SELECT SUM(_discount_multi_extendedprice_) from test_lineitem_mv` doesn't have an output name, the output visible fields won't provide the name thus will fail.

Reply:
> Given query `SELECT SUM(_discount_multi_extendedprice_) from test_lineitem_mv` doesn't have an output name, the output visible fields won't provide the name thus will fail.

Interesting, are you saying users would need to have it as 
SELECT SUM(_discount_multi_extendedprice_) as _discount_multi_extendedprice_ from test_lineitem_mv to make it work? 
My setup is broken so can not try it at the moment.

Reply:
not necessary; it's just we don't have an output field name; but the field is still there.